usr/lib/common/mech_des3.c

```diff
@@ -2796,16 +2792,44 @@ CK_RV ckm_des3_key_gen(STDLL_TokData_t *tokdata, TEMPLATE *tmpl)
     local_attr->pValue = (CK_BYTE *) local_attr + sizeof(CK_ATTRIBUTE);
     *(CK_BBOOL *) local_attr->pValue = TRUE;
 
-    template_update_attribute(tmpl, value_attr);
-    template_update_attribute(tmpl, key_type_attr);
-    template_update_attribute(tmpl, class_attr);
-    template_update_attribute(tmpl, local_attr);
+    rc = template_update_attribute(tmpl, value_attr);
+    if (rc != CKR_OK) {
+        TRACE_ERROR("template_update_attribute failed\n");
+        return rc;
+    }
+    value_attr = NULL;
+    rc = template_update_attribute(tmpl, key_type_attr);
+    if (rc != CKR_OK) {
+        TRACE_ERROR("template_update_attribute failed\n");
+        return rc;
+    }
+    key_type_attr = NULL;
+    rc = template_update_attribute(tmpl, class_attr);
+    if (rc != CKR_OK) {
+        TRACE_ERROR("template_update_attribute failed\n");
+        return rc;
+    }
+    class_attr = NULL;
+    rc = template_update_attribute(tmpl, local_attr);
+    if (rc != CKR_OK) {
+        TRACE_ERROR("template_update_attribute failed\n");
+        return rc;
```

Comment:
memory leak.  Use `goto error`.

Reply:
good catch! Thanks for reading so carefully!
Will fix this and all the others as well.